metalearners/cross_fit_estimator.py

```diff
@@ -337,8 +338,28 @@ def predict_proba(
             oos_method=oos_method,
         )
 
-    def score(self, X, y, sample_weight=None, **kwargs):
-        raise NotImplementedError()
+    def score(
+        self,
+        X: Matrix,
+        y: Vector,
+        is_oos: bool,
+        oos_method: OosMethod | None = None,
+        sample_weight: Vector | None = None,
+    ) -> float:
+        """Return the coefficient of determination of the prediction if the estimator is
+        a regressor or the mean accuracy if it is a classifier."""
+        if self._estimator_type == "classifier":
+            return accuracy_score(
+                y, self.predict(X, is_oos, oos_method), sample_weight=sample_weight
+            )
+        elif self._estimator_type == "regressor":
+            return r2_score(
```

Comment:
Any particular reason why you chose R2, rather than e.g. the RMSE?

Reply:
Just because it looks like it is the convention in ``sklearn``, see [this](https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.LGBMRegressor.html#lightgbm.LGBMRegressor.score) and [this](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html#sklearn.linear_model.LinearRegression.score).
I think they do it because the R2 has as known range whether the RMSE depends a lot on the data but I just followed their rule, R2 for regressors and accuracy for classifiers.